community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/ExecutionPlanBuilder.scala

```diff
@@ -82,10 +87,21 @@ class ExecutionPlanBuilder(graph: GraphDatabaseService, statsDivergenceThreshold
       def isStale(lastTxId: () => Long, statistics: GraphStatistics) = fingerprint.isStale(lastTxId, statistics)
 
       def run(queryContext: QueryContext, kernelStatement: KernelStatement,
-                       planType: ExecutionMode, params: Map[String, Any]): InternalExecutionResult =
-        compiledPlan.executionResultBuilder(kernelStatement, graph, planType, params, new CompletionListener {
-          override def complete(success: Boolean) = queryContext.close(success)
-        })
+              planType: ExecutionMode, params: Map[String, Any]): InternalExecutionResult = {
```

Comment:
We should probably rename `planType` to something with `mode`. `executionMode`, `mode`, `queryMode`?